### PR TITLE
only reply if response topic set

### LINKED ISSUE #142: Interference/problems with miniconf and other mqtt paths
Using the latest master branch with eg. Stabilizer dual-iir (aka using vanilla Stabilizer master branch, just changing the miniconf dep to the github master and minimq to 0.6.2) leads to the following issues:

Miniconf seems to be confused with messages that are not intended for it. I get the following debug prints:
```
INFO - Starting
INFO - EUI48: 80-34-28-1f-59-65
INFO - setup() complete
WARN - Network link DOWN
WARN - Network link UP
INFO - MQTT connected, subscribing to settings
INFO - Unknown Miniconf command: /alive
INFO - Unknown Miniconf command: /telemetry
```

If I publish onto a miniconf setting I get `WARN - Failed to build response message`.

The miniconf settings are applied at the device and I also see the mqtt messages like `/alive` on the host.

### PR DESCRIPTION
**Pull request by jordens**
* [x] only try to build response if ResponseTopic is set
* [x] should subscribe to `/settings/#` and `/list` explicitly, not to `/#` (which captures telemetry, alive, and maybe even response topics, log etc as well)

closes #142

### CODE REVIEW

File: src/mqtt_client/mod.rs [hunk 1]
```diff
@@ -96,21 +96,20 @@ enum Command<'a> {
...
-                if value.is_empty() {
-                    Ok(Command::Get { path })
-                } else {
-                    Ok(Command::Set { path, value })
+
+        if path == "list" {
+            Ok(Command::List)
+        } else {
+            match path.split_once('/') {
+                Some(("settings", path)) => {
```
Comment (ryan-summers) [inline src/mqtt_client/mod.rs:104]: Does this properly check that `path` is not None?

Comment (jordens) [reply src/mqtt_client/mod.rs:104]: In this arm `path` is always `&str`.

**Review by ryan-summers — APPROVED** [review]

File: src/mqtt_client/mod.rs [hunk 2]
```diff
@@ -442,20 +446,10 @@ where
         F: FnMut(&str, &mut Settings, &Settings) -> Result<(), E>,
         E: AsRef<str>,
     {
-        let Self {
```
Comment (ryan-summers) [inline src/mqtt_client/mod.rs:445]: Huh, I wonder what changed that made this no longer necessary. I'm glad we can get rid of it in any case!

**Review by ryan-summers — CHANGES_REQUESTED** [review]
Can you please add relevant CHANGELOG entries for the fact that we fixed an issue about subscriptions and that we now only respond if there's a response topic?

File: src/mqtt_client/mod.rs (hunk 2 above)
Comment (jordens) [reply src/mqtt_client/mod.rs:445]: Yeah. This became unnecessary a while ago. It now accurately captures individual items in structs (and not the entire struct). Also https://doc.rust-lang.org/1.33.0/book/ch18-03-pattern-syntax.html#legacy-patterns-ref-and-ref-mut

**Review by ryan-summers — APPROVED** [review]

**Pull request merged**